.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,44 @@
+name: Build and Test Workflow
```

Comment:
Can we call this from `ci.yaml` like the rest of them? By the looks of it it is only about contracts, so it should be named `contracts-build-and-test.yaml` to fit in with its brethren. Then it will also benefit from the conditional execution tags.